src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -140,10 +140,18 @@ public void InitializeService(ServiceHost serviceHost)
                     Version = Version.Parse(serverInfo.ServerVersion),
                     Platform = hostInfo.Platform
                 };
-
                 var db = SqlAssessmentService.GetDatabaseLocator(server, connection.Database);
-                var connectionString = ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
-                var results = await GetAssessmentItems(server, connectionString);
+                var connectionStrings = new List<string>();
+                if (parameters.Databases != null) 
+                {
+                    foreach (string database in parameters.Databases)
+                    {
+                        connInfo.ConnectionDetails.DatabaseName = database;
+                        connectionStrings.Add(ConnectionService.BuildConnectionString(connInfo.ConnectionDetails));
+                    }
+                }
+                string[] assessmentConnectionStrings = connectionStrings.ToArray();
```

Comment:
should this entire section be skipped parameters.Databases == null

Reply:
are you saying I should remove the `if (parameters.Databases != null)`?

Reply:
should lines 154 - 157 be executed if the database count is 0, i.e. assessmentConnectionStrings is empty
if no, can they go inside the if check.